client/src/lib/src/BitbakeDriver.ts

```diff
@@ -11,30 +11,52 @@ import { type BitbakeSettings, loadBitbakeSettings } from './BitbakeSettings'
 /// This class is responsible for wrapping up all bitbake classes and exposing them to the extension
 export class BitbakeDriver {
   bitbakeSettings: BitbakeSettings = { pathToBitbakeFolder: '', pathToBuildFolder: '', pathToEnvScript: '', workingDirectory: '', commandWrapper: '' }
+  bitbakeActive = false
 
   loadSettings (settings: any, workspaceFolder: string = '.'): void {
     this.bitbakeSettings = loadBitbakeSettings(settings, workspaceFolder)
     logger.debug('BitbakeDriver settings updated: ' + JSON.stringify(this.bitbakeSettings))
   }
 
+  private async waitForBitbakeToFinish (): Promise<void> {
+    await new Promise<void>((resolve) => {
+      const interval = setInterval(() => {
+        if (!this.bitbakeActive) {
```

Comment:
We are guaranteed that bitbakeActive is put back to false at some point by
```Typescript
child.on('close', () => {
```
From the ChildProcess doc:
```
The 'close' event will always emit after ['exit'](https://nodejs.org/api/child_process.html#event-exit) was already emitted, or ['error'](https://nodejs.org/api/child_process.html#event-error) if the child failed to spawn.
```

If the command is very long (which is possible), yes, it needs to wait for it to complete. All commands are displayed to the user though through terminal tasks at this point, and he can kill them. The only exceptions are the show-layers and show-appends from the BitbakeProjectScanner. Those should not be particularly slow except if the process is frozen. It can happen in rare cases, like a broken mount point. However in those cases, we don't lose much since the system is apparently broken. I could take note on that though. There are alternatives or ways to report suspiciously long invocations. I'll open a ticket for that.